main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/madlambda/bwc/infix"
+)
+
+func main() {
+	for {
+		fmt.Printf("bwc> ")
+		var line [256]byte
```

Comment:
Although I admit that a line that is bigger than 256 bytes is really hard to exist I dislike arbitrary limits like this. Can't we use the [bufio.Scanner](https://golang.org/pkg/bufio/#NewScanner) here ? I think the code would get smaller and we remove the arbitrary limit on the line length (or perhaps it is not arbitrary).

Reply:
hahaha sorry for that
This main file was just my 'test binary' to play in the acme win. I dont like bufio.Scanner because it's a little monster and hard to configure.. then I made something fast to use. 
In our case, the statements could span multiple lines, then we will need to continue to read lines from input while parser notifies/reports 'premature eof' error (maybe using error types or err behaviours).